marklogic-data-hub-glue-connector/src/main/java/com/marklogic/hub/cloud/aws/glue/Writer/MarkLogicDataWriterFactory.java

```diff
@@ -15,31 +15,43 @@
  */
 package com.marklogic.hub.cloud.aws.glue.Writer;
 
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.hub.HubClient;
+import com.marklogic.hub.impl.HubConfigImpl;
 import org.apache.spark.sql.catalyst.InternalRow;
 import org.apache.spark.sql.sources.v2.writer.DataWriter;
 import org.apache.spark.sql.sources.v2.writer.DataWriterFactory;
 import org.apache.spark.sql.types.StructType;
 
-import java.util.HashMap;
 import java.util.Map;
+import java.util.Properties;
+
+public class MarkLogicDataWriterFactory extends LoggingObject implements DataWriterFactory<InternalRow> {
 
-public class MarkLogicDataWriterFactory implements DataWriterFactory<InternalRow> {
-    private Map<String, String> map;
     private StructType schema;
-    public MarkLogicDataWriterFactory(Map<String, String> map, StructType schema) {
-        this.map = map;
+    private HubClient hubClient;
+    private Map<String, String> params;
+
+    /**
+     * @param params a map of parameters containing both DHF-supported properties (most likely prefixed with ml* or
+     *               hub*) and connector-specific properties. The DHF-supported properties will be used to construct a
+     *               HubClient for communicating with MarkLogic.
+     * @param schema
+     */
+    public MarkLogicDataWriterFactory(Map<String, String> params, StructType schema) {
+        this.params = params;
         this.schema = schema;
+
+        Properties props = new Properties();
```

Comment:
@anu3990  @SameeraPriyathamTadikonda  This is the main change here. We should configure the connector via DHF-supported properties - e.g. mlHost, mlUsername, mlPassword. Those can then be passed into HubConfigImpl.withProperties to construct a HubClient, which knows how to talk to each app server in DHF. 

The connector can still have its own set of properties, such as "prefix" and "batchSize".